database/postgres/postgres.go

```diff
@@ -332,21 +333,23 @@ func (p *Postgres) Drop() error {
 	return nil
 }
 
-func (p *Postgres) ensureVersionTable() error {
-	// check if migration table exists
-	var count int
-	query := `SELECT COUNT(1) FROM information_schema.tables WHERE table_name = $1 AND table_schema = (SELECT current_schema()) LIMIT 1`
-	if err := p.conn.QueryRowContext(context.Background(), query, p.config.MigrationsTable).Scan(&count); err != nil {
-		return &database.Error{OrigErr: err, Query: []byte(query)}
-	}
-	if count == 1 {
-		return nil
+func (p *Postgres) ensureVersionTable() (err error) {
+	if err = p.Lock(); err != nil {
+		return err
 	}
 
-	// if not, create the empty migration table
-	query = `CREATE TABLE "` + p.config.MigrationsTable + `" (version bigint not null primary key, dirty boolean not null)`
-	if _, err := p.conn.ExecContext(context.Background(), query); err != nil {
+	defer func() {
+		if e := p.Unlock(); err == nil {
+			err = e
+		} else if e != nil {
+			err = multierror.Append(err, e)
```

Comment:
It's unclear to me if `multierror.Append()` will work correctly if the first arg is `nil`.
I think it's safer to use logic like:
```golang
if e:= p.Unlock(); e != nil {
    if err == nil {
        err = e
    } else {
        err = multierror.Append(err, e)
    }
}
```

Reply:
> Also, the else if condition contains a hidden check for both the err and e variables.

What is a hidden check?

Reply:
the `else if e != nil` is actually `if err == nil && e != nil` but isn't as readable since it's a single conditional checking 2 variables in different branches.

Reply:
That's what an else branch always does. In the extreme case, if we want to be super explicit, we'd do this:

```go
if x == 2 {
  // it's 2
} else if x != 2 {
  // not 2
}
```

But this is just style preference. So I'll re-write to your preferred approach.